moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -981,6 +1004,40 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(
     RCLCPP_ERROR(LOGGER, "It looks like the planner did not set the group the plan was computed for");
     return false;
   }
+
+  // Validate scaling
+  double velocity_scaling_factor = 1.0;
+  if (max_velocity_scaling_factor > 0.0 && max_velocity_scaling_factor <= 1.0)
+  {
+    velocity_scaling_factor = max_velocity_scaling_factor;
+  }
+  else if (max_velocity_scaling_factor == 0.0)
+  {
+    RCLCPP_DEBUG(LOGGER, "A max_velocity_scaling_factor of 0.0 was specified, defaulting to %f instead.",
+                 velocity_scaling_factor);
+  }
+  else
+  {
+    RCLCPP_WARN(LOGGER, "Invalid max_velocity_scaling_factor %f specified, defaulting to %f instead.",
+                max_velocity_scaling_factor, velocity_scaling_factor);
+  }
+
+  double acceleration_scaling_factor = 1.0;
+  if (max_acceleration_scaling_factor > 0.0 && max_acceleration_scaling_factor <= 1.0)
+  {
+    acceleration_scaling_factor = max_acceleration_scaling_factor;
+  }
+  else if (max_acceleration_scaling_factor == 0.0)
+  {
+    RCLCPP_DEBUG(LOGGER, "A max_acceleration_scaling_factor of 0.0 was specified, defaulting to %f instead.",
```

Comment:
Just saw that you took this part of the code from the existing implementation. @henningkayser or @AndyZe can you answer my question here?

Reply:
Just adding my $0.02 about why this is happening. My guess is that when filling out a motionrequest, it is helpful to not always have to define the scaling factors to be nonzero (since they will be zero by default) but rather to just have the algorithm to default to 1. It also a good failsafe since it's a safe bet that if a user forgot to define them that the default limits should be used without any scaling.
In regards to why DEBUG is used, probably because relying on the default values is commonly done so people didn't want the terminal/log file to be spammed with these warnings haha

Reply:
merged this condition with the other invalid condition in latest commit so no more debug log